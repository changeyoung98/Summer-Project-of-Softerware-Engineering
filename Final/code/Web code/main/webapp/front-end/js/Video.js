import React from 'react'
import {Button,message} from 'antd'
import Demo from './demo'

const warning =()=> {
  message.warning('Waiting for selecting a camera');
};

const warning1 = () =>{
  message.warning('Please wait for a while')
};
const success = ()=>{
  message.success("Cut the picture and save it !")
};


class Video extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      now_camera:"None",
      url: null,
      history: [],
      instance: [],
      currentTime: null,
      state: 0,
      camera:
        [
          {
            instance: "rtmp://192.168.1.100:1935/oflaDemo",
            history: "http://221.228.226.23/11/t/j/v/b/tjvbwspwhqdmgouolposcsfafpedmb/sh.yinyuetai.com/691201536EE4912BF7E4F1E2C67B8119.mp4"
          },
          {
            instance: "rtmp://192.168.1.127:1935/oflaDemo",
            history: "http://localhost:8080/video/test.mp4"
          }]
    };
    this.getVideo = this.getVideo.bind(this);
    this.getVideoh = this.getVideoh.bind(this);
    this.get = this.get.bind(this);
    this.gets = this.gets.bind(this);
  }

  getVideo(index) {
    if(index===this.state.now_camera) {
      this.setState({
        state: 1,
        url: this.state.camera[index-1],
        now_camera: "None",
        history: [],
        instance: [],
      });
    }
    else{
      this.setState({
        state: 1,
        url: this.state.camera[index-1],
        now_camera: index,
        history: [],
        instance: [],
      });
    }
  }

  getVideoh() {
    console.log(this.state.url.history);
    let tem = [];
    if (this.state.state === 1) {
      tem.push(
        <video id="video1" className="video-js" controls preload="auto" width="320" height="240" data-setup="{}">
          <source src={this.state.url.history} type="video/mp4"/>
        </video>
      );
    }
    else {
      warning();
    }
    this.setState({
      history: tem,
    })
  }

  gets() {
    if (this.state.state === 1 && this.state.history !== []) {
      warning1();
      var myPlayer = document.getElementById('video1');
      let time = myPlayer.currentTime;
      this.setState({
        currentTime: time,
      }, () => {
        let history_url = this.state.url.history;
        let path = history_url.split("/");
        let video_name = path[path.length-1];
        $.ajax({
          type: "GET",
          url: 'http://localhost:8080/videos/cut',
          data: {
            url: video_name,
            time: this.state.currentTime,
          },
          dataType: "json",
          success: function (data) {
            console.log(this.state.currentTime);
            console.log("success");
            if (JSON.parse(data) === 1) {
              success();
            }
          }.bind(this)
        });
      });
      this.props.handleTime(this.state.currentTime);
    }
    else {
      warning();
    }
    /*let time = this.state.currentTime;
    let result = time.split(":");
    result = result.reverse();
    let sec = 0;
    for (let i = 0;i<result.length;i++){
      sec += parseInt(result[i])*(60**i);
    }
    let secs = sec.toString();
    $.ajax({
      type: "GET",
      url: 'http://localhost:8081/video/cut',
      data: {
        url:this.state.historyUrl,
        time:secs,
      },
      dataType: "json",
      success: function (data) {
        console.log(secs);
        console.log("success");
      }.bind(this)
    });*/
  }

  get() {
    let tem = [];
    let url = this.state.url;
    if(this.state.state===1) {
      tem.push(
        <Demo src={url.instance}/>
      );
    }
    else{
      warning();
    }
    this.setState({
      instance: tem,
    })
    /* if (this.state.state === 1) {
      let tem = 0;
      $.ajax({
        type: "GET",
        url: 'http://localhost:8081/videos/Icut',
        data: {
          url: this.state.url.instance,
        },
        dataType: "json",
        success: function (data) {
          tem = JSON.parse(data);
          console.log("success");
        }.bind(this)
      });
    }
    else {
      warning();
    }
    */
  }

  render() {
    return (
      <div>
        <img src="./floor1.jpg" height="300" width="500" align="center" useMap="#mymap"/>
        <map name="mymap">
          <area shape="rect" coords="0,0,100,100" alt="camera1" onClick={() => {
            this.getVideo(1)
          }}/>
          <area shape="rect" coords="240,0,320,100" alt="camera2" onClick={() => {
            this.getVideo(2)
          }}/>
        </map>

        <div id="player">
          <div>
            <h3>Now Camera : {this.state.now_camera}</h3>
          </div>
          <div><Button type="primary" onClick={this.get}>Real-Time Video</Button></div>
          <div>{
            this.state.instance
          }
          </div>
          </div>
        <div className="input">
          <Button type="primary" onClick={this.getVideoh}>Historical Video</Button>
          <Button type="primary" onClick={this.gets}>Cut</Button>
        </div>
        <div id="playerh">
          {this.state.history}
        </div>
      </div>
    )
  }
}

export default Video;
