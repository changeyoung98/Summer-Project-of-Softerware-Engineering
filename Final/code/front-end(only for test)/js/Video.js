import React from 'react'
import {Button,message} from 'antd'
import Play from './Play'

const warning =()=> {
  message.warning('Waiting for selecting a camera');
};

const warning1 = () =>{
  message.warning('Please wait for a while')
};
const success = ()=>{
  message.success("Cut the picture and save it !")
};

class Video extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      now_camera: null,
      url: null,
      history: [],
      instance:[],
      currentTime: null,
      state: 0,
      camera:
        [
          {
            instance: "rtmp://pull-g.kktv8.com/livekktv/100987038 ",
            history: "http://221.228.226.23/11/t/j/v/b/tjvbwspwhqdmgouolposcsfafpedmb/sh.yinyuetai.com/691201536EE4912BF7E4F1E2C67B8119.mp4"
          },
          {
            instance: "rtmp://192.168.1.127:1935/oflaDemo/stream15308",
            history: "http://localhost:8081/video/test.mp4"
          }]
    };
    this.getVideo = this.getVideo.bind(this);
    this.getVideoh = this.getVideoh.bind(this);
    this.get = this.get.bind(this);
    this.gets = this.gets.bind(this);
  }

  getVideo(index) {
    let tem=[];
    let url = this.state.camera[index];
    tem.push(
      <Play url={url.instance}/>
    );
    this.setState({
      state: 1,
      url: this.state.camera[index],
      now_camera: index,
      history:[],
      instance:tem,
    });
  }

  getVideoh() {
 //   console.log(this.state.url.history);
    let tem = [];
    if (this.state.state === 1) {
      tem.push(
        <video id="video1" className="video-js" controls preload="auto" width="320" height="240" data-setup="{}">
          <source src={this.state.url.history} type="video/mp4"/>
        </video>
      );
    }
    else {
      warning();
    }
    this.setState({
      history: tem,
    })
  }

  gets() {
    if (this.state.state === 1 && this.state.history !== []) {
      warning1();
      var myPlayer = document.getElementById('video1');
      let time = myPlayer.currentTime;
      this.setState({
        currentTime: time,
      }, () => {
        $.ajax({
          type: "GET",
          url: 'http://localhost:8081/videos/cut',
          data: {
            url: this.state.url.history,
            time: this.state.currentTime,
          },
          dataType: "json",
          success: function (data) {
            console.log(this.state.currentTime);
            console.log("success");
            if (JSON.parse(data) === 1) {
              success();
            }
          }.bind(this)
        });
      });
      this.props.handleTime(this.state.currentTime);
    }
    else {
      warning();
    }
    /*let time = this.state.currentTime;
    let result = time.split(":");
    result = result.reverse();
    let sec = 0;
    for (let i = 0;i<result.length;i++){
      sec += parseInt(result[i])*(60**i);
    }
    let secs = sec.toString();
    $.ajax({
      type: "GET",
      url: 'http://localhost:8081/video/cut',
      data: {
        url:this.state.historyUrl,
        time:secs,
      },
      dataType: "json",
      success: function (data) {
        console.log(secs);
        console.log("success");
      }.bind(this)
    });*/
  }

  get() {
    if (this.state.state === 1) {
      let tem = 0;
      $.ajax({
        type: "GET",
        url: 'http://localhost:8081/videos/Icut',
        data: {
          url: this.state.url.instance,
        },
        dataType: "json",
        success: function (data) {
          tem = JSON.parse(data);
     //     console.log("success");
        }.bind(this)
      });
    }
    else {
      warning();
    }
  }

  render() {
    return (
      <div>
        <img src="./floor1.jpg" height="300" width="500" align="center" useMap="#mymap"/>
        <map name="mymap">
          <area shape="rect" coords="0,0,100,100" alt="camera1" onClick={() => {
            this.getVideo(1)
          }}/>
          <area shape="rect" coords="240,0,320,100" alt="camera2" onClick={() => {
            this.getVideo(0)
          }}/>

        </map>

        <div id="player">
          <div>
            <h3>Camera {this.state.now_camera}</h3>
          </div>
          <div><Button type="primary" onClick={this.get}>Start</Button></div>
          <div>
            {this.state.instance}
          </div>
        </div>
        <div>
          <div>
            History Video
          </div>
          <Button type="primary" onClick={this.getVideoh}>Click</Button>
          <Button type="primary" onClick={this.gets}>Cut</Button>
          <p>{this.state.currentTime}</p>
        </div>
        <div id="playerh">
          {this.state.history}
        </div>
      </div>


    )
  }
}

export default Video;
