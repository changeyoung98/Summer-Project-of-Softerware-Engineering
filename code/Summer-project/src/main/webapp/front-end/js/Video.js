import React from 'react'
import {Button} from 'antd'
import Demo from './demo'

class Video extends React.Component{
    constructor(props){
        super(props);
        this.state={
          camera:0,
          url:"rtmp://live.hkstv.hk.lxdns.com/live/hks",
          history:[],
          currentTime:null,
          historyUrl:null,
        };
        this.getVideo = this.getVideo.bind(this);
        this.getVideoh=this.getVideoh.bind(this);
        this.get = this.get.bind(this);
        this.handleTime = this.handleTime.bind(this);
        this.gets = this.gets.bind(this);
        this.handleUrl = this.handleUrl.bind(this);
    }

    handleUrl(url){
      this.setState({
        historyUrl:url,
      })
    }

    handleTime(time){
      this.setState({
        currentTime:time,
      })
    }

    getVideo(){
        this.setState({
          camera:1,
        })
    }
    getVideoh(){
      let tem = [];
      tem.push(
        <Demo handleTime={this.handleTime} handleUrl={this.handleUrl}/>
      );
      this.setState({
        history:tem,
      })
    }

    gets(){
      let time = this.state.currentTime;
      let result = time.split(":");
      result = result.reverse();
      let sec = 0;
      for (let i = 0;i<result.length;i++){
        sec += parseInt(result[i])*(60**i);
      }
      let secs = sec.toString();
      $.ajax({
        type: "GET",
        url: 'http://localhost:8081/video/cut',
        data: {
          url:this.state.historyUrl,
          time:secs,
        },
        dataType: "json",
        success: function (data) {
          console.log(secs);
          console.log("success");
        }.bind(this)
      });
    }

  get(){
    let tem = 0;
    $.ajax({
      type: "GET",
      url: 'http://localhost:8081/video/Icut',
      data: {
        url:this.state.url,
      },
      dataType: "json",
      success: function (data) {
        tem = JSON.parse(data);
        console.log("success");
      }.bind(this)
    });
  }

  render(){
        return(
        <div>
            <img src="./floor1.jpg"  height="300" width="500" align="center" useMap="#mymap"/>
                <map name="mymap">
                    <area shape = "rect" coords="0,0,100,100" alt="camera1" onClick={this.getVideo} />

                </map>

            <div id="player">
              <div><Button type="primary" onClick={this.get}>Start</Button></div>
              <div>
                {
                  this.state.camera===0?null:
                    <iframe
                      src="/front-end/demo.html"
                      width="320px"
                      height="240px"
                      allowFullScreen="true"
                    >nice</iframe>
                }
              </div>
            </div>
            <div>
                <div>
                History Video
                </div>
                <Button type="primary" onClick={this.getVideoh}>Click</Button>
                <Button type="primary" onClick={this.gets}>Cut</Button>
              <p>{this.state.currentTime}</p>
            </div>
            <div id="playerh">
              {this.state.history}
            </div>
        </div>


        )
    }
}

export default Video;
