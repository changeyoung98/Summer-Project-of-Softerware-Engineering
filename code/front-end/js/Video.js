import React from 'react'
import {Button} from 'antd'

class Video extends React.Component{
    constructor(props){
        super(props);
        this.getVideo = this.getVideo.bind(this);
    }

    getVideo(){
        jwplayer("player").setup({
            flashplayer:'player.swf',
            image:"preview.jpg",
             file:"stream15308",
            streamer:"rtmp://localhost/oflaDemo"
        });
        console.log("haha");
    }

    render(){
        return(
            <div id="player">
                <Button type="primary" onClick={this.getVideo}>Click</Button>

            </div>



        )
    }
}

export default Video;