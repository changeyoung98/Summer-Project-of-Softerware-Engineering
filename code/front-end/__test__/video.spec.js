/**
 * Created by hp on 2018/7/24.
 */
import React from 'react'
import Video from '../js/Video'
import {Button} from 'antd'
import toJson from 'enzyme-to-json';
import { shallow, configure,render,mount } from 'enzyme'
import  Adapter from 'enzyme-adapter-react-14';

configure({ adapter: new Adapter() });
const setup = () => {
    // 模拟 props

    const props = {
        // Jest 提供的mock 函数
        state : {
            now_camera: null,
            url: null,
            history: [],
            instance:[],
            currentTime: null,
            state: 0,
            camera:
                [
                    {
                        instance: "rtmp://pull-g.kktv8.com/livekktv/100987038 ",
                        history: "http://221.228.226.23/11/t/j/v/b/tjvbwspwhqdmgouolposcsfafpedmb/sh.yinyuetai.com/691201536EE4912BF7E4F1E2C67B8119.mp4"
                    },
                    {
                        instance: "rtmp://192.168.1.127:1935/oflaDemo/stream15308",
                        history: "http://localhost:8081/video/test.mp4"
                    }]
        },
        getVideo : jest.fn(),
        getVideoh: jest.fn(),
        get:jest.fn(),
        gets:jest.fn(),
    };
    const wrapper = shallow(<Video {...props}/>)

    return {
        props,
        wrapper,
    }
};

describe('Video render', () => {
    const {wrapper, props} = setup();
    it('basic uses', () => {
        const wrapperS = render(
            <Video {...props} />
        );
        expect(toJson(wrapperS)).toMatchSnapshot();
    });

    it('Button Component should render', () => {
        //.find(selector) 是 Enzyme shallow Rendering 提供的语法, 用于查找节点
        // 详细用法见 Enzyme 文档 http://airbnb.io/enzyme/docs/api/shallow.html
        expect(wrapper.find('Button').length).toBe(3);
    });


    it('1 Component should click', () => {
        const buttonDom = wrapper.find('Button').at(0);
        buttonDom.simulate('click');
        expect(props.get).toBeCalled;
    });
    it('2 Component should click', () => {
        const buttonDom = wrapper.find('Button').at(1);
        buttonDom.simulate('click');
        expect(props.getVideoh).toBeCalled;
    });
    it('3 Component should click', () => {
        const buttonDom = wrapper.find('Button').at(2);
        buttonDom.simulate('click');
        expect(props.gets).toBeCalled;
    });
    it(' area Component should click', () => {
        const buttonDom = wrapper.find('area').at(0);
        buttonDom.simulate('click');
        expect(props.getVideo(1)).toBeCalled;
    });
    it('A Component should click', () => {
        const buttonDom = wrapper.find('Button').at(1);
        buttonDom.simulate('click');
        expect(props.getVideo(0)).toBeCalled;
    });
});