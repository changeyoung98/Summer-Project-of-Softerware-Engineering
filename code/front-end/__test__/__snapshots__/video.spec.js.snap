// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Video render basic uses 1`] = `
<div>
  <img
    height="300"
    src="./floor1.jpg"
    usemap="#mymap"
    width="500"
  />
  <map
    name="mymap"
  >
    <area
      alt="camera1"
      coords="0,0,100,100"
      shape="rect"
    />
    <area
      alt="camera2"
      coords="240,0,320,100"
      shape="rect"
    />
  </map>
  <div
    id="player"
  >
    <div>
      <h3>
        Camera 
      </h3>
    </div>
    <div>
      <button
        class="ant-btn ant-btn-primary"
        type="button"
      >
        <span>
          Start
        </span>
      </button>
    </div>
    <div />
  </div>
  <div>
    <div>
      History Video
    </div>
    <button
      class="ant-btn ant-btn-primary"
      type="button"
    >
      <span>
        Click
      </span>
    </button>
    <button
      class="ant-btn ant-btn-primary"
      type="button"
    >
      <span>
        Cut
      </span>
    </button>
    <p />
  </div>
  <div
    id="playerh"
  />
</div>
`;
